app/controllers/assignment_invitations_controller.rb

```diff
@@ -52,22 +52,24 @@ def accept
   def setup; end
 
   def setupv2
-    render status: 404 unless import_resiliency_enabled?
+    return not_found unless import_resiliency_enabled?
   end
 
   # rubocop:disable MethodLength
   def create_repo
     if import_resiliency_enabled?
       job_started = false
       if current_invitation.accepted? || current_invitation.errored?
+        assignment_repo = AssignmentRepo.find_by(assignment: current_assignment, user: current_user)
+        assignment_repo&.destroy if assignment_repo
```

Comment:
This worries me just a little bit, if some bug in the current implementation or a bug introduced at a later date allows `create_repo` to be fired after a student has started working on their assignment, there's a chance we might destroy their in-progress assignment and all the work they pushed up to GitHub.

Do you have any ideas for how we might be able to be more confident here? One idea I came up with was querying for the number of commits on the repo if it exists, and only destroying if it's 0. it's a bit of overhead, but the case where there's an existing assignment_repo isn't the common path, and I'd rather pay a little bit for the confidence it'll give us. Nobody likes seeing their work destroyed :/

Reply:
I like the idea of querying the repo for the number of commits 👌 That extra layer of check is worth it.

Reply:
Any chance there could be more than 0 commits if creating the repo fails?